web/core/src/jsMain/kotlin/androidx/compose/web/attributes/PredefinedAttrValues.kt

```diff
@@ -1,26 +1,101 @@
 package org.jetbrains.compose.web.attributes
 
-sealed class InputType(val typeStr: String) {
-    object Button : InputType("button")
-    object Checkbox : InputType("checkbox")
-    object Color : InputType("color")
-    object Date : InputType("date")
-    object DateTimeLocal : InputType("datetime-local")
-    object Email : InputType("email")
-    object File : InputType("file")
-    object Hidden : InputType("hidden")
-    object Month : InputType("month")
-    object Number : InputType("number")
-    object Password : InputType("password")
-    object Radio : InputType("radio")
-    object Range : InputType("range")
-    object Search : InputType("search")
-    object Submit : InputType("submit")
-    object Tel : InputType("tel")
-    object Text : InputType("text")
-    object Time : InputType("time")
-    object Url : InputType("url")
-    object Week : InputType("week")
+import org.w3c.dom.events.Event
+
+sealed class InputType<T>(val typeStr: String) {
+
+    object Button : InputType<Unit>("button") {
+        override fun inputValue(event: Event) = Unit
+    }
+
+    object Checkbox : InputType<Boolean>("checkbox") {
+        override fun inputValue(event: Event): Boolean {
+            return event.target?.asDynamic()?.checked?.unsafeCast<Boolean>() ?: false
+        }
+    }
+
+    object Color : InputType<String>("color") {
+        override fun inputValue(event: Event) = valueAsString(event)
```

Comment:
why we need to override this all the time rather than have it somewhere higher in a hierarchy?

Reply:
added an improvement for this.
let me know what you think